doc/SONIC_Test_Ingress_Discards_HLD.md

```diff
@@ -0,0 +1,999 @@
+
+##### For IP drop reasons:
+```show interface counters rif``` - check ```RX_ERR```
+
+##### For ACL drop reasons:
+```aclshow -a``` - check ```PACKETS COUNT```
+
+#### Scope
+The purpose of test cases is to verify that:
+- appropriate packet drop counters trigger on SONIC system on expected value
+- making sure that specific traffic drops correctly, according to sent packet and configured packet discards
+
+#### Supported topologies:
+```
+t0
+t1
+t1-lag
+ptf32
+```
+
+#### Discard groups covered by test cases
+Please refer to the test case for detailed description.
+
+| Test case ID| Drop reason | Group type|
+|-------------|-------------|-----------|
+| 1 | SMAC and DMAC are equal |Ethernet |
+| 2 | Not allowed VLAN TAG| Ethernet|
+| 3 | Multicast SMAC | Ethernet|
+| 4 | Reserved DMAC | Ethernet|
+| 5 | Loop-back filter | IP|
+| 6 | Packet exceed router interface MTU| IP|
+| 7 | Time To Live (TTL) Expired | IP|
+| 8 | Discard at router interface for non-routable packets | IP|
+| 9 | Absent IP header | IP|
+| 10 | Broken IP header - header checksum or IPver or IPv4 IHL too short | IP|
+| 11 | Unicast IP with multicast DMAC or broadcast DST MAC | IP|
+| 12 | DST IP is loopback address | IP|
+| 13 | SRC IP is loopback address | IP|
+| 14 | SRC IP is multicast address | IP|
+| 15 | SRC IP address is in class E | IP|
+| 16 | SRC IP address is not specified | IP|
+| 17 | DST IP address is not specified | IP|
+| 18 | SRC IP address is link-local | IP|
+| 19 | DST IP address is link-local | IP|
+| 20 | ACL SRC IP DROP| IP|
+| 21 | ERIF interface disabled | IP|
+
+#### Related DUT CLI commands
+| **Command**                                                      | **Comment** |
+|------------------------------------------------------------------|-------------|
+| show interfaces counters              | Check ```RX_DRP```                     |
+| counterpoll rif enable                | Enable RIF counters                    |
+| show interface counters rif           | Check ```RX_ERR```                     |
+| aclshow -a                            | Check ```PACKETS COUNT```              |
+| sonic-clear counters                  | Clear counters                         |
+
+Listed above commands can be different for different vendors. So there should be created appropriate parser for CLI command per vendor.
+There will be created a file with constant CLI commands per vendor. Then this mapping will be used by test cases automatically based on vendor info got from DUT.
+
+For example:
+```
+- Vendor X:
+    "get_l2_pkt_drops": "show interfaces counters"
+    "get_l3_pkt_drops": "show interfaces counters rif"
+    "get_acl_pkt_drops": "aclshow -a"
+    "parser": "path_to_the_parser_py"
+...
+```
+
+##### Work need to be done based on this case
+- Create yml file which defines mapping of vendor to CLI commands for getting specific drop counters
+- Create helper, CLI command parser per vendor - python file which contains three functions:
```

Comment:
Is the yml file necessary or can we just provide a python script that implements the three functions you've described?

Reply:
Simplified approach. As different vendors can't have different SONiC CLI, CLI is common. However internal counters calculation can be different. So it is enough to know some vendor identifier to determine whether current DUT supports combined drop counters or not.
So platform name is selected for such reason.

Reply:
So with this change we will just have a method that parses the L2 output, a method that parses the L3 output, and a method that parses the ACL counter output?

Reply:
Yes